command_line/find_rotation_axis.py

```diff
@@ -0,0 +1,496 @@
+    diff_phil = parser.diff_phil.as_str()
+    if diff_phil:
+        logger.info("The following parameters have been modified:\n%s", diff_phil)
+
+    reflections, experiments = reflections_and_experiments_from_files(
+        params.input.reflections, params.input.experiments
+    )
+
+    # Check the models and data
+    nexp = len(experiments)
+    if nexp > 1:
+        logger.info(
+            "Only the first experiment will be used to determine oscillation and current rotation axis"
+        )
+        if len(experiments.goniometers()) > 1:
+            sys.exit("Multiple experiments must share a goniometer model")
+    if nexp == 0 or len(reflections) == 0:
+        parser.print_help()
+        return
+    if len(reflections) > 1:
+        sys.exit("Only one reflections list can be imported at present")
+    reflections = reflections[0]
+    expt = experiments[0]
+    wavelength = expt.beam.get_wavelength()
+    rotx, roty, _ = expt.goniometer.get_rotation_axis()
+    azimuth_current = np.degrees(np.arctan2(roty, rotx))
+    arr = extract_spot_data(reflections, experiments, params.max_two_theta)
+
+    if params.azimuth is not None:
+        azimuth_current = params.azimuth
+
+    azimuth_opposite = azimuth_current + 180
+
+    if params.opposite:
+        azimuth_current = azimuth_opposite
+
+    if azimuth_current > 180:
+        azimuth_current -= 360
+
+    if azimuth_opposite > 180:
+        azimuth_opposite -= 360
+
+    logger.info(f"Wavelength: {wavelength:.5f} Ångström")
+    logger.info(f"Azimuth (current): {azimuth_current:.5f} degrees")
+    logger.info(f"                 {np.radians(azimuth_current):.5f} radians")
+
+    hist_bins = 1000, 500
+
+    if params.view:
+        azimuth_final = azimuth_current
+    elif params.optimise:
+        global xvals
+        global vvals
+        xvals = []
+        vvals = []
+
+        azimuth_global = azimuth_local = azimuth_fine = 0
+
+        if not params.global_search:
+            azimuth_tmp = azimuth_global = azimuth_current
+        else:
+            logger.info("Performing global search")
+            azimuth_tmp = 0
+            azimuth_global = azimuth_tmp = optimise(
+                arr, azimuth_tmp, wavelength, plusminus=180, step=5, hist_bins=hist_bins
+            )
+
+        logger.info("Performing local search")
+        azimuth_local = azimuth_tmp = optimise(
+            arr, azimuth_tmp, wavelength, plusminus=5, step=1, hist_bins=hist_bins
+        )
+
+        logger.info("Performing fine search")
+        azimuth_fine = azimuth_tmp = optimise(
+            arr, azimuth_tmp, wavelength, plusminus=1, step=0.1, hist_bins=hist_bins
+        )
+
+        azimuth_final = azimuth_tmp
+
+        logger.info("---")
+        logger.info(f"Best azimuth (global search): {azimuth_global:.3f}")
+        logger.info(f"Best azimuth (local search): {azimuth_local:.3f}")
+        logger.info(f"Best azimuth (fine search): {azimuth_fine:.3f}")
```

Comment:
Just a note (don't hold up the PR for this): logging isn't print - Things like this should be a single log call.

Reply:
Understood. This comes from a find-and-replace starting from the `print`s in `edtools.find_rotation_axis`. Happy to make this a single call before merge.